updater/osv/osv_test.go

```diff
@@ -138,18 +141,509 @@ func TestParse(t *testing.T) {
+				},
+				FixedInVersion: "2.1.16",
+			},
+			{
+				Name:    "test1",
+				Updater: "test",
+				Range: &claircore.Range{
+					Lower: claircore.FromSemver(semver.MustParse("2.2.0")),
+					Upper: claircore.FromSemver(semver.MustParse("2.2.10")),
+				},
+				FixedInVersion: "2.2.10",
+			},
+			{
+				Name:    "test1",
+				Updater: "test",
+				Range: &claircore.Range{
+					Lower: claircore.FromSemver(semver.MustParse("2.3.0")),
+					Upper: claircore.Version{
+						Kind: "semver",
+						V:    [10]int32{65535, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+					},
+				},
+				FixedInVersion: "",
+			},
+		},
+	},
+	{
+		// In this situation we're just expecting the last one.
+		name: "two_consecutive_introduced_invalid",
+		ad: &advisory{
+			ID: "test1",
+			Affected: []affected{
+				{
+					Package: _package{
+						Ecosystem: "go",
+						Name:      "something",
+					},
+					Ranges: []_range{
+						{
+							Type: "SEMVER",
+							Events: []rangeEvent{
+								{
+									Introduced: "2.2.0",
+								},
+								{
+									Introduced: "2.3.0",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		expectedVulns: []claircore.Vulnerability{
+			{
+				Name:    "test1",
+				Updater: "test",
+				Range: &claircore.Range{
+					Lower: claircore.FromSemver(semver.MustParse("2.3.0")),
+					Upper: claircore.Version{
+						Kind: "semver",
+						V:    [10]int32{65535, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+					},
+				},
+				FixedInVersion: "",
+			},
+		},
+	},
+	{
+		name: "ecosystem_multi",
+		ad: &advisory{
+			ID: "test1",
+			Affected: []affected{
+				{
+					Package: _package{
+						Ecosystem: "PyPI",
+						Name:      "cherrypy",
+					},
+					Ranges: []_range{
+						{
+							Type: "ECOSYSTEM",
+							Events: []rangeEvent{
+								{
+									Introduced: "0",
+								},
+								{
+									Fixed: "2.1.1",
+								},
+								{
+									Introduced: "3.0",
+								},
+								{
+									Fixed: "3.0.2",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		expectedVulns: []claircore.Vulnerability{
+			{
+				Name:           "test1",
+				Updater:        "test",
+				Range:          nil,
+				FixedInVersion: "fixed=2.1.1",
+			},
+			{
+				Name:           "test1",
+				Updater:        "test",
+				Range:          nil,
+				FixedInVersion: "fixed=3.0.2&introduced=3.0",
+			},
+		},
+	},
+	{
+		name: "ecosystem_unfixed",
+		ad: &advisory{
+			ID: "test1",
+			Affected: []affected{
+				{
+					Package: _package{
+						Ecosystem: "PyPI",
+						Name:      "cherrypy",
+					},
+					Ranges: []_range{
+						{
+							Type: "ECOSYSTEM",
+							Events: []rangeEvent{
+								{
+									Introduced: "3.0",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		expectedVulns: []claircore.Vulnerability{
+			{
+				Name:           "test1",
+				Updater:        "test",
+				Range:          nil,
+				FixedInVersion: "introduced=3.0",
+			},
+		},
+	},
+	{
+		name: "same package different ranges",
```

Comment:
sorry I meant for `ecosystem` since we already have one for `semver` a few tests above ^

Reply:
ahh